src/MongoDB.Driver/MongoClient.cs

```diff
@@ -223,13 +231,37 @@ public sealed override IMongoDatabase GetDatabase(string name, MongoDatabaseSett
             return CreateDatabaseNamesCursor(databases);
         }
 
+        /// <inheritdoc />
+        public sealed override IAsyncCursor<string> ListDatabaseNames(
+            IClientSessionHandle session,
+            ListDatabaseNamesOptions options,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var listDatabasesOptions = new ListDatabasesOptions { NameOnly = true };
+            if (options != null)
+            {
+                listDatabasesOptions.AuthorizedDatabases = options.AuthorizedDatabases;
+                listDatabasesOptions.Filter = options.Filter;
+            }
```

Comment:
Create a helper method for the above lines to avoid code duplication:

    var listDatabasesOptions = CreateListDatabasesOptionsFromListDatabaseNamesOptions(options);

Reply:
Helper method:

    private ListDatabasesOptions CreateListDatabasesOptionsFromListDatabaseNamesOptions(ListDatabaseNamesOptions options)
    {
        var listDatabasesOptions = new ListDatabasesOptions { NameOnly = true };
        if (options != null)
        {
            listDatabasesOptions.AuthorizedDatabases = options.AuthorizedDatabases;
            listDatabasesOptions.Filter = options.Filter;
        }
        return listDatabasesOptions;
    }

Reply:
Done.